test/controllers/ChangeSuccessControllerSpec.scala

```diff
@@ -99,22 +103,25 @@ class ChangeSuccessControllerSpec extends ControllerBaseSpec with MockContactPre
 
     "both expected session keys are populated" when {
 
-      "the user is a principal entity" should {
+      "the user is a principal entity"  when {
 
-        lazy val result: Future[Result] = {
-          getMockContactPreference(vrn)(Future.successful(Right(ContactPreference("DIGITAL"))))
-          mockGetEmailVerificationStatus(Future(Some(true)))
-          controller.websiteAddress(request.withSession(
-            prepopulationWebsiteKey -> "", websiteChangeSuccessful -> "true"
-          ))
-        }
+        "the contactPrefMigrationEnabled feature switch is turned on" should {
 
-        "return 200" in {
-          status(result) shouldBe Status.OK
-        }
+          lazy val result: Future[Result] = {
+            mockGetCustomerInfo(vrn)(Right(fullCustomerInfoModel))
+            mockGetEmailVerificationStatus(Future(Some(true)))
+            controller.websiteAddress(request.withSession(
+              prepopulationWebsiteKey -> "", websiteChangeSuccessful -> "true"
+            ))
+          }
+
+          "return 200" in {
+            status(result) shouldBe Status.OK
+          }
 
-        "not call the VatSubscription service" in {
-          verify(mockVatSubscriptionService, times(0)).getCustomerInfo(any())(any(), any())
+          "not call the VatSubscription service" in {
+            verify(mockVatSubscriptionService, times(0)).getCustomerInfo(any())(any(), any())
```

Comment:
I'm a bit confused as to why this test is passing as you _are_ calling the `getCustomerInfo` function in the vat subscription service. It's being mocked on line 111 at least. Same as all the other tests you have for "feature switch is turned off".

These tests will change a bit after a refactor of the controller, but something to still keep in mind.